sopel/plugin.py

```diff
@@ -120,6 +120,13 @@ class CapabilityNegotiation(enum.Enum):
     """
 
 
+if TYPE_CHECKING:
+    CapabilityHandler = Callable[
+        [Tuple[str, ...], SopelWrapper, bool],
+        CapabilityNegotiation,
+    ]
```

Comment:
Nitpick: Is the `TYPE_CHECKING` guard necessary? I ask because it seems plausible that this could be used by consumers who want to generate their handlers, although this is absolutely a fringe edge case.

Reply:
Yes it is, sadly: ` SopelWrapper` cannot be imported in `sopel.plugin`, because `sopel.bot` defines `SopelWrapper` and imports `sopel.plugin` (for a single constant that I really need to move out...).

So, yeah, to prevent circular import, I have to. At least for now.